### PR TITLE
PLATFORM-4216: limit elasticsearch cardinality for mediawiki index

### PR DESCRIPTION
**Pull request by themech**
* LUA traces are the biggest offenders, pack those into a single field
* use json_encode for fields PHP arrays (like headers)
* unify field names
* resolve type conflicts

### CODE REVIEW

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 1]
```diff
@@ -766,7 +766,7 @@ private static function getMediaDetailWithSizeFallback(
 				\Wikia\Logger\WikiaLogger::instance()->notice(
 					'ArticleAsJson - Media height was empty - fallback to fallbackSize',
 					[
-						'media_details' => $mediaDetail,
+						'media_details' => json_encode( $mediaDetail ),
```
Comment (mszabo-wikia) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php:769]: Can we make this sort of preprocessing automatic? In the logging method, we could iterate over the context and encode all array keys.

Comment (themech) [reply extensions/wikia/ArticleAsJson/ArticleAsJson.class.php:769]: technically it is possible. but still there are low-cardinality array-type fields which I use for filtering, like `@context.edit_status.is_ok `, `@context.ex.message` `@context.reqStatus.is_ok` to name a few.
Now, we could have a rule that we encode arrays having more than 4 elements. But now we get into a situation that if for the same field we have arrays of different sizes, then sometimes you encode it as array and sometimes as strings. And due to type conflicts ES will not index those incompatible values.
So I think patching the high-cardinality arrays at the source is actually easier.

Comment (mszabo-wikia) [reply extensions/wikia/ArticleAsJson/ArticleAsJson.class.php:769]: Makes sense, thanks!

File: extensions/wikia/Forum/RelatedForumDiscussionController.class.php [hunk 2]
```diff
@@ -39,8 +39,8 @@ public static function purgeCache( $threadId ) {
 			$wikiaPage = WikiPage::newFromID( $id );
 			if ( $wikiaPage ) {
 				$wikiaPage->doPurge();
+				self::logError( "Found a null related wikipage on thread purge", [ "articleId" => $id, "threadId" => $threadId ] );
```
Comment (Grunny) [inline extensions/wikia/Forum/RelatedForumDiscussionController.class.php]: I believe this shouldn't have been moved from the `else` below?

Comment (mszabo-wikia) [reply extensions/wikia/Forum/RelatedForumDiscussionController.class.php]: We might want to consider removing it outright as I don't think anybody is monitoring this error anyways.

**Review by Grunny — APPROVED** [review]

**Pull request merged**